src/logic/safe/hooks/useSafeScheduledUpdates.tsx

```diff
@@ -17,25 +16,20 @@ export const useSafeScheduledUpdates = (safeLoaded: boolean, safeAddress?: strin
     // has to run again
     let mounted = true
     const fetchSafeData = async (address: string): Promise<void> => {
-      await batch(async () => {
+      batch(async () => {
         await Promise.all([
-          dispatch(fetchEtherBalance(address)),
           dispatch(fetchSafeTokens(address)),
           dispatch(fetchTransactions(address)),
           dispatch(fetchCollectibles(address)),
           dispatch(checkAndUpdateSafe(address)),
         ])
       })
-
-      if (mounted) {
-        timer.current = window.setTimeout(() => {
-          fetchSafeData(address)
-        }, TIMEOUT * 3)
-      }
     }
 
     if (safeAddress && safeLoaded) {
-      fetchSafeData(safeAddress)
+      if (mounted && !timer.current) {
```

Comment:
is it possible to write the same condition in one level?

Reply:
Yes, there was already an inner if and I didn't thought about it